tools/link.py

```diff
@@ -792,8 +792,8 @@ def phase_linker_setup(options, linker_args):  # noqa: C901, PLR0912, PLR0915
   if settings.WASM_ESM_INTEGRATION:
     diagnostics.warning('experimental', '-sWASM_ESM_INTEGRATION is still experimental and not yet supported in browsers')
     default_setting('MODULARIZE', 'instance')
-    if options.oformat != OFormat.MJS:
-      exit_with_error('WASM_ESM_INTEGRATION is only compatible with EM module output format')
+    if not settings.EXPORT_ES6:
+      exit_with_error('WASM_ESM_INTEGRATION requires EXPORT_ES6')
```

Comment:
which cases do we implicitly set dependent flags? I guess that has already happened before we get to this point?

Reply:
Yes we deliberately do this early up on line 754